presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayUnionFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.operator.aggregation.TypedSet;
+import com.facebook.presto.operator.scalar.annotations.ScalarFunction;
+import com.facebook.presto.operator.scalar.annotations.TypeParameter;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.SqlType;
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+import it.unimi.dsi.fastutil.longs.LongSet;
+
+import javax.annotation.Nullable;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+@ScalarFunction("array_union")
+@Description("Union elements of the two given arrays")
+public final class ArrayUnionFunction
+{
+    private ArrayUnionFunction() {}
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public static Block union(
+            @TypeParameter("E") Type type,
+            @Nullable @SqlType("array(E)") Block leftArray,
```

Comment:
I don't think skipping `NULL` s is the right semantics. Generally, scalar functions return `NULL` when any of their inputs are `NULL` because `NULL` represents an unknown value, and so in this case we don't know what's in the union when we don't know whats in the array on one side.